src/meta/meta_data.cpp

```diff
@@ -479,6 +479,37 @@ void app_state_helper::on_init_partitions()
     restore_states.resize(owner->partition_count);
 }
 
+void app_state_helper::reset_manual_compact_status()
+{
+    for (auto &cc : contexts) {
+        for (auto &r : cc.serving) {
+            r.compact_status = manual_compaction_status::IDLE;
+        }
+    }
+}
+
+bool app_state_helper::get_manual_compact_progress(/*out*/ int32_t &progress) const
+{
+    int32_t total_replica_count = owner->partition_count * owner->max_replica_count;
+    int32_t finish_count = 0, idle_count = 0;
+    for (const auto &cc : contexts) {
+        for (const auto &r : cc.serving) {
+            if (r.compact_status == manual_compaction_status::IDLE) {
+                idle_count++;
+            } else if (r.compact_status == manual_compaction_status::FINISHED) {
+                finish_count++;
+            }
+        }
+    }
+    // all replica of all partitions are idle
+    if (idle_count == total_replica_count) {
+        progress = 0;
+        return false;
+    }
+    progress = finish_count * 100 / total_replica_count;
```

Comment:
Does total_replica_count always greater than 0?

Reply:
total_replica_count = partition_count * replica_count, both of them should always be greater than 0.

Reply:
So in some error case, total_replica_count may be 0. So I think it's better to add a judgement here.

Reply:
I think it is almost a impossible error case, should coredump because of `total_replica_count=0`. I think it's okay to maintain the code or add an assert sentence for it. How about your suggestion?

Reply:
assert is better, in my opinion :)

Reply:
Done